src/main/java/useCases/createCorporatePage/PageGenerateInteractor.java

```diff
@@ -0,0 +1,59 @@
+package useCases.createCorporatePage;
+import entity.CorporatePage;
+import entity.Page;
+
+public class PageGenerateInteractor implements PageGenerateInputBoundary {
+
+    final PageGenerateDsGateway pageGenerateDsGateway;
+    final PageGenerateOutputBoundary pageGenerateOutputBoundary;
+    final PageFactory pageFactory;
+
+    public PageGenerateInteractor(PageGenerateDsGateway pageGenerateDsGateway,
+                                  PageGenerateOutputBoundary pageGenerateOutputBoundary, PageFactory pageFactory){
+        this.pageGenerateDsGateway = pageGenerateDsGateway;
+        this.pageGenerateOutputBoundary = pageGenerateOutputBoundary;
+        this.pageFactory = pageFactory;
+    }
+
+    public PageGenerateResponseModel create(PageGenerateRequestModel requestModel){
+
+        if (pageGenerateDsGateway.existsByPageLabel(requestModel.getInputLabel())){
+            return pageGenerateOutputBoundary.prepareFailView("Page already exists.");
+        }
+
+        Page page = pageFactory.create(requestModel.getInputLabel(), requestModel.getUser());
+
+        PageGenerateResponseModel pageResponseModel;
+
+        if (requestModel.getPageType().equals("Corporate")){
```

Comment:
We can use an enum class with type `COPORATE` and `USER` so it's less error prone. Your `pageType` will then only take two values.